source/plugins/go/input/lib/applicationinsights.go

```diff
@@ -0,0 +1,339 @@
+	for k, v := range customProperties {
+		telemetryProps[k] = v
+	}
+
+	// add passed-in dimensions if any
+	for k, v := range properties {
+		telemetryProps[k] = v
+	}
+
+	if telemetryClient != nil {
+		event := appinsights.NewEventTelemetry(eventName)
+		event.Properties = telemetryProps
+		telemetryClient.Track(event)
+		aiLogger.Printf("AppInsights Custom Event %s sent successfully\n", eventName)
+	}
+}
+
+func SendExceptionTelemetry(errorStr string, properties map[string]string) {
+	if customProperties["DockerVersion"] == "" {
+		getContainerRuntimeInfo()
+	}
+
+	telemetryProps := make(map[string]string)
+
+	// add common dimensions
+	for k, v := range customProperties {
+		telemetryProps[k] = v
+	}
+
+	// add passed-in dimensions if any
+	for k, v := range properties {
+		telemetryProps[k] = v
+	}
+
+	if telemetryClient != nil {
+		exception := appinsights.NewExceptionTelemetry(errorStr)
+		exception.Properties = telemetryProps
+		telemetryClient.Track(exception)
+		aiLogger.Printf("AppInsights Exception Telemetry put successfully into the queue")
+	}
+}
+
+func SendTelemetry(pluginName string, properties map[string]string) {
+	if customProperties["DockerVersion"] == "" {
+		getContainerRuntimeInfo()
+	}
+
+	customProperties["Computer"] = properties["Computer"]
+	if v, ok := properties["addonTokenAdapterImageTag"]; ok && v != "" {
+		customProperties["addonTokenAdapterImageTag"] = v
+	}
+
+	sendHeartBeatEvent(pluginName)
+	sendLastProcessedContainerInventoryCountMetric(pluginName, properties)
+}
+
+func SendMetricTelemetry(metricName string, metricValue float64, properties map[string]string) {
+	if metricName == "" {
+		log.Println("SendMetricTelemetry: metricName is missing")
+		return
+	}
+
+	if customProperties["DockerVersion"] == "" {
+		getContainerRuntimeInfo()
+	}
+
+	telemetryProps := make(map[string]string)
+
+	// add common dimensions
+	for k, v := range customProperties {
+		telemetryProps[k] = v
+	}
+
+	// add passed-in dimensions if any
+	for k, v := range properties {
+		telemetryProps[k] = v
+	}
+
+	if telemetryClient != nil {
+		metric := appinsights.NewMetricTelemetry(metricName, metricValue)
+		metric.Properties = telemetryProps
+		telemetryClient.Track(metric)
+		aiLogger.Printf("AppInsights metric Telemetry %s put successfully into the queue\n", metricName)
+	}
+}
+
+func SendException(err interface{}) {
+	if telemetryClient != nil {
+		telemetryClient.TrackException(err)
+	}
+}
+
+func getWorkspaceId() string {
+	workspaceID := os.Getenv("WSID")
+	if workspaceID == "" {
+		aiLogger.Printf("Exception in AppInsightsUtility: getWorkspaceId - WorkspaceID either nil or empty")
+	}
+	return workspaceID
+}
+
+func getWorkspaceCloud() string {
+	workspaceDomain := os.Getenv("DOMAIN")
+	workspaceCloud := "AzureCloud"
+	switch strings.ToLower(workspaceDomain) {
+	case "opinsights.azure.com":
+		workspaceCloud = "AzureCloud"
+	case "opinsights.azure.cn":
+		workspaceCloud = "AzureChinaCloud"
+	case "opinsights.azure.us":
```

Comment:
All the new AI telemetry code needs to be updated to match the changes in this PR - https://github.com/microsoft/Docker-Provider/commit/7559a07b5d28d8e223a665945fe139a53359d109